specs/v2-sketch.yaml

```diff
@@ -0,0 +1,365 @@
+        parallelism:
+          type: number
+          format: integer
+        dev_admin:
+          type: boolean
+          description: Not sure
+          deprecated: true
+        in_beta_program:
+          type: boolean
+        bitbucket_authorized:
+          type: boolean
+          description: This is a remnant of not being a platform, do we really want to expose it?
+          deprecated: true
+        enrolled_betas:
+          type: array
+          items:
+            type: string
+        analytics_id:
+          type: string
+          format: uuid
+          description: can we call this "id", please?
+        student:
+          type: boolean
+          default: false
+        privacy-optout:
+          type: boolean
+          description: Opt out of analytics trackers
+#concurrency
+#   Appears unused
+#
+#encrypted-heroku-api-key
+#   Nope, not including credentials, encrypted or not.
+#
+#plans
+#   Not sure what these should be
+        containers:
+          type: number
+          format: integer
+        num_projects_followed:
+          type: number
+          format: integer
+        days_left_in_trial:
+          type: number
+          format: integer
+        all_emails:
+          type: array
+          example: ["username@example.com"]
+          items:
+            type: string
+        selected_email:
+          type: string
+          example: "username@example.com"
+        avatar_url:
+          type: string
+          example: "https://avatars.example.com/369ea08d7172bfc72207e2ae1c1b17ef"
+# github_id
+# bitbucket
+# bitbucket-authorized
+# pusher-id
+#   These all feel far too specific, should be rolled into "identities", right?
+#
+# github-oauth-scopes
+#   don't wanna include GitHub oauth scopes
+#
+# heroku-api-key
+#   Nope, not including credentials, obfuscated or not
+        identities:
+          type: object
+          additionalProperties:
+            type: object
+            properties:
+              avatar_url:
+                type: string
+                example: "https://avatars.example.com/369ea08d7172bfc72207e2ae1c1b17ef"
+              external_id:
+                anyOf:
+                  - type: string
+                  - type: number
+                example: 1234567
+              id:
+                anyOf:
+                  - type: string
+                  - type: number
+                example: 1234567
+              name:
+                type: string
+                example: "User Name"
+              user?:
+                type: boolean
+              domain:
+                type: string
+                example: "github.com"
+              type:
+                type: string
+                example: "github"
+              authorized?:
+                type: boolean
+              provider_id:
+                type: string
+                format: uuid
+              login:
+                type: string
+                example: "username"
+    Actor:
```

Comment:
How will actor's that are not circleci user's be represented?

Reply:
The `login` and `name` fields will be populated from the hook data, the `id` will be empty (this needs documenting). That's quite coupled to our existing model, my belief is that the concepts of `login` and/or `name` will exist for other sources of identity but I'd like to hear your thoughts.

Being explicit and including a flag for known users is possible too.

It would be nice to be able to link to the identity from the external source, however, GitHub (at least) don't provide a stable URL for users in the face of renames.

I don't have a good sense yet of whether the actor information should be immutable or not. I.e. if a customer signs up should we return what we now know for workflows they triggered when they were not a customer?

Reply:
Will we return login and name for things like API triggers? Are those denormalized in any way that should be called out?